src/script/components/InputBar/ReplyBar.tsx

```diff
@@ -0,0 +1,107 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {FC} from 'react';
+import Icon from 'Components/Icon';
+import Image from 'Components/Image';
+import RestrictedVideo from 'Components/asset/RestrictedVideo';
+import ParticipantMicOnIcon from 'Components/calling/ParticipantMicOnIcon';
+
+import {renderMessage} from 'Util/messageRenderer';
+import {t} from 'Util/LocalizerUtil';
+import {ContentMessage} from '../../entity/message/ContentMessage';
+
+interface ReplyBarProps {
+  replyMessageEntity: ContentMessage | null;
+  onCancel: () => void;
+}
+
+const ReplyBar: FC<ReplyBarProps> = ({replyMessageEntity, onCancel}) => {
+  const replyAsset = replyMessageEntity?.assets?.()[0];
+
+  return (
+    <div className="input-bar__reply" data-uie-name="input-bar-reply-box">
+      <button type="button" className="button-reset-default" onClick={onCancel} data-uie-name="do-close-reply-box">
+        <Icon.Close fill="white" />
+      </button>
+
+      <div className="input-bar__reply__body">
+        <div className="input-bar__reply__vert-bar"></div>
+
+        <div className="input-bar__reply__text">
+          <div className="input-bar__reply__sender-name">
+            <span data-uie-name="label-name-reply-box">{replyMessageEntity?.headerSenderName()}</span>
+
+            {replyMessageEntity?.was_edited() && <Icon.Edit data-uie-name="message-edited-reply-box" />}
+          </div>
+
+          {replyAsset?.isText() && (
+            <div
+              className="input-bar__reply__message input-bar__reply__message__text"
+              data-uie-name="media-text-reply-box"
+              dangerouslySetInnerHTML={{__html: renderMessage(replyAsset.text, null, replyAsset.mentions())}}
+            />
+          )}
+
+          {replyAsset?.isImage() && (
+            <div data-uie-name="media-picture-reply-box">
+              <Image
+                className="bar__reply__message input-bar__reply__message__image"
+                asset={replyAsset.resource()}
+                isQuote
+              />
+            </div>
+          )}
+
+          {replyAsset?.isVideo() && (
+            <div data-uie-name="media-video-reply-box">
+              <RestrictedVideo className="input-bar__reply__message" showMessage={false} isSmall />
+            </div>
+          )}
+
+          {replyAsset?.isAudio() && (
+            <div className="input-bar__reply__message" data-uie-name="media-audio-reply-box">
+              <ParticipantMicOnIcon className="input-bar__reply__icon" />
+
+              <span>{t('replyAudioMessage')}</span>
+            </div>
+          )}
+
+          {replyAsset?.isFile() && (
+            <div className="input-bar__reply__message" data-uie-name="media-file-reply-box">
+              <Icon.File className="input-bar__reply__icon" />
```

Comment:
all the icons should have an alternative text for screen readers only, and icons should have aria-hidden={true}

Reply:
this one `aria-hidden={true}` is handled by `SVGProvider` check out `components/Icon.tsx` file 😄